src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -276,15 +278,20 @@ class CsvToHttp extends ApiHandler {
           // match if the string starts with: ${...}
           if (element.match(REGEX_MATCH_VARIABLE_STRING)) {
             const headerToGet = element.match(REGEX_GET_VARIABLE)
-            headerToGet.forEach((header) => {
-              const response = CsvToHttp.convertToCorrectType(csvRowInJson[header], 'string')
+            // The regex must match with only one value and return an array with one element
+            if (headerToGet.length !== 1) {
+              object.error.push(`Regex doesn't match only with one value (tried element: ${element})`)
+            }
```

Comment:
Next if must be 'else if' instead. Indeed : if headerToGet.length == 0, you push the error, but then the next if is executed.